src/Microsoft.ML.Data/Transforms/FeatureContributionCalculationTransform.cs

```diff
@@ -39,8 +46,23 @@ namespace Microsoft.ML.Data
     ///         FastForest, FastTree, LightGbm
     ///     Ranking:
     ///         FastTree, LightGbm
-    ///
+    /// </para>
+    /// <para>
+    /// For linear models, the contribution of a given feature is equal to the product of feature value times the corresponding weight. Similarly,
+    /// for Generalized Additive Models (GAM), the contribution of a feature is equal to the shape function for the given feature evaluated at
+    /// the feature value.
+    /// </para>
+    /// <para>
+    /// For tree-based models, the contribution of a feature is equal to the change in score produced by exploring the opposite sub-tree every time a decision
+    /// node for the given feature is encountered. Consider a simple case with a single decision tree that has a decision node for the binary feature F1.
+    /// Given an example that has feature F1 equal to true, we can calculate the score it would have obtained if we chose the subtree corresponding to
+    /// the feature F1 being equal to false while keeping the other features constant. The contribution of feature F1 for the given example is the difference
+    /// between the original score and the score obtained by taking the opposite decision at the node corresponding to feature F1. This algorithm extends
+    /// naturally to models with many decision trees.
```

Comment:
This is a very mechanical explanation of the model we use to calculate the contribution of features in a tree model. I'd add a bit of "how to interpret" to this. For example, one way you can interpret the we calculate contributions is that we are calculating which splits in the tree have the most impact, and assigning those values to the feature determining the split.